rna/input_output.py

```diff
@@ -0,0 +1,250 @@
+                    cell type was measured,
+                N_cell types,
+                N_markers (=N_features),
+                dict: cell type name -> cell type index,
+                dict: cell type index -> cell type name,
+                dict: cell type index -> N_measurements for cell type
+
+    """
+
+    df, rv = read_df(filename, binarize, number_of_replicates)
+
+    label_encoder = LabelEncoder()
+    if single_cell_types:
+        single_cell_types = list(set(single_cell_types))
+        single_cell_types.append('Skin.penile')
+        label_encoder.fit(single_cell_types)
+    else:
+        if not ground_truth_known:
+            raise ValueError('if no cell types are provided, ground truth should be known')
+        # if not provided, learn the cell types from the data
+        all_celltypes = np.array(df.index)
+        for celltype in all_celltypes:
+            # TODO: How does this work if single_cell_types is None?
+            if celltype not in single_cell_types and celltype!='Skin.penile':
+                raise ValueError('unknown cell type: {}'.format(celltype))
+
+        label_encoder.fit(all_celltypes)
+
+    n_celltypes_with_penile = len(single_cell_types)
+    n_features = len(df.columns)
+    n_per_celltype = dict()
+
+    X_single=[]
+    if ground_truth_known:
+        for celltype in list(single_cell_types) + ['Skin.penile']:
+            data_for_this_celltype = np.array(df.loc[celltype])
+
+            if type(rv) == pd.core.frame.DataFrame:
+                rvset_for_this_celltype = np.array(rv.loc[celltype]).flatten()
+            # TODO: Currently does not work
+            elif type(rv) == list:
+                rvset_for_this_celltype = rv[rv[:, 1] == celltype, 0]
+
+            n_full_samples, X_for_this_celltype = get_data_for_celltype(celltype, data_for_this_celltype,
+                                                                      indices_per_replicate, rvset_for_this_celltype)
+
+            for repeated_measurements in X_for_this_celltype:
+                X_single.append(repeated_measurements)
+            n_per_celltype[celltype] = n_full_samples
+
+        y_nhot_single = np.zeros((len(X_single), n_celltypes_with_penile))
+        end = 0
+        for i, celltype in enumerate(list(single_cell_types) + ['Skin.penile']):
+            i_celltype = label_encoder.transform([celltype])
+            begin = end
+            end = end + n_per_celltype[celltype]
+            y_nhot_single[begin:end, i_celltype] = 1
+
+    else:
+        n_full_samples, X_single = get_data_for_celltype('Unknown', np.array(df), indices_per_replicate, rv)
+        y_nhot_single=None
+
+    X_single = np.array(X_single)
+
+    assert X_single.shape[0] == y_nhot_single.shape[0]
+
+    return X_single, y_nhot_single, n_celltypes_with_penile, n_features, n_per_celltype, \
+           label_encoder, list(df.columns), list(df.index)
+
+
+def read_mixture_data(filename, n_celltypes, label_encoder, binarize=True):
```

Comment:
waren er goede redenen om dit niet in de bovenstaande methode te vlechten? Laat voor nu maar staan maar dan kunnen we dat nog eens doen..